hom/utils.py

```diff
@@ -40,37 +40,41 @@ def get_category(guild: discord.Guild, channel_name: str) -> t.Optional[discord.
 
 def get_channel_by_id(
     bot: commands.Bot, guild_id: int, channel_id: int, channel_type: str = "text"
-) -> t.Optional[discord.guild.GuildChannel]:
+) -> t.Optional[discord.TextChannel]:
     if guild_id not in [guild.id for guild in bot.guilds]:
         return None
 
     if not (guild := bot.get_guild(guild_id)):
         return None
 
-    return discord.utils.find(
+    channel = discord.utils.find(
         lambda c: c.id == channel_id and str(c.type).lower() == channel_type,
         guild.channels,
     )
 
+    return t.cast(discord.TextChannel, channel) if channel else None
+
 
 def get_channel_by_name(
     bot: commands.Bot | discord.Client,
     guild_id: int,
     channel_name: str,
     channel_type: str = "text",
-) -> t.Optional[discord.guild.GuildChannel]:
+) -> t.Optional[discord.TextChannel]:
```

Comment:
This isn't accurate if we ever use a different "channel_type". Might be addressed in a later commit but it seems like maybe the more correction option for us to do would be to remove the channel_type and strictly use this for getting text channels and get_category() for categories.

All that said, not sure if "-> t.Optional[discord.TextChannel]:" literally means it may not be a text channel and that we are defaulting the type to a text channel because the channel_type parameter is defaulted to "text".

Reply:
This function no longer exists also. But if the ID in the environment is wrong we will never find the channel. That's why its optional.